src/Analyzers/Core/CodeFixes/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -276,12 +277,12 @@ ImmutableArray<CodeAction> NestByOverload()
                 builder.Add(codeAction);
             }
 
-            return builder.ToImmutableAndClear();
+            return builder.MoveToImmutable();
         }
 
         ImmutableArray<CodeAction> NestByCascading()
         {
-            using var _ = ArrayBuilder<CodeAction>.GetInstance(capacity: 2, out var builder);
+            using var builder = TemporaryArray<CodeAction>.Empty;
```

Comment:
tiny array cases could just use temporaryarray.  this guy is then cancellation-safe.  teh values are placed on the stack, and copied at the end.  since this is only for <4 element cases, this is basically free.